driver-core/src/test/resources/unified-test-format/transactions/java-only-mongos-unpin.json

```diff
@@ -0,0 +1,96 @@
+{
```

Comment:
This may be controversial: The spec test doesn't cover all the paths in the Java driver to test this unpinning logic.  That's because there are two paths for operation execution, one for reads and one for writes.  The spec test exercises the write path.  I added this test to execute the read path as well.

Reply:
Agreed, lets make it less controversial and add it to repo. Other drivers may benefit from it also.

Reply:
https://jira.mongodb.org/browse/DRIVERS-1685

Reply:
Thank you for creating the ticket.